### ISSUE TITLE
Migrate examples/groups.jl to src/modeltools/transformations.jl

### ISSUE THREAD

**Issue by jpfairbanks** [jpfairbanks/SemanticModels.jl#139]
Now that the examples have stabilized, I think we can migrate this code into the module.

**Comment by jpfairbanks** [comment]
While doing that, we should add this Problem type which will work for every code

```julia
issome(x) = !ismissing(x) && !isa(x, Nothing)
head(x::Expr) = x.head
head(n::LineNumberNode) = nothing
isblock(x) = head(x) == :block
isfunc(x) = head(x) ==:function
or(f::Function, g::Function) = x->(f(x) || g(x))

mutable struct SimpleProblem
    expr::Expr
    imports::Any
    blocks::Vector{Expr}
    functions::Vector{Expr}
end

function model(::Type{SimpleProblem}, expr::Expr)
    body = expr.args[2].args[3]
    statements = body.args
    imports = filter(issome, 
        map(x-> if head(x) == :using 
                   return x
                else 
                   return nothing
                end,
            statements)
    )
    blocks = filter(or(isblock, isfunc), statements)
    funcs = filter(isfunc, statements)
    return SimpleProblem(expr, imports, blocks, funcs)
end
```

**Comment by jpfairbanks** [comment]
Closed by #143
